tools/single_use/remove_publication_users_edit_perm_rel_duplicates.php

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env php
+<?php declare(strict_types=1);
+error_reporting(E_ALL);
+require_once __DIR__ . "/../generic_includes.php";
+
+echo
+"\nThis script deletes duplicate rows in publication_users_edit_perm_rel
+to make sure the patch '2020-01-07-publication_users_edit_perm_rel_pk.sql' will execute successfully.\n\n";
+
+echo "Searching for duplicates...\n\n";
+
+$query = "SELECT *
+FROM publication_users_edit_perm_rel
+GROUP BY
+    PublicationID,
+    UserID
+HAVING
+    (COUNT(PublicationID) > 1) AND
+    (COUNT(UserID) > 1);";
+
+$result = $DB->pselect($query, array());
+
+if (count($result) > 0) {
+    echo "Deleting duplicates: \n";
+} else {
+    echo "No duplicate found. \n";
+}
+
+foreach ($result as $row) {
+    foreach ($row as $key => $value) {
+        echo $key . ':' . $value . '    ';
+    }
+
+    $query = "DELETE
+        FROM publication_users_edit_perm_rel
+        WHERE PublicationID = $row[PublicationID]
+        AND UserID = $row[UserID]
+        LIMIT 1";
```

Comment:
The issue with this `LIMIT 1` clause is that if there are multiple duplicates, the script does not delete them!

For example:
Before running script –
<img width="875" alt="Screen Shot 2021-03-18 at 1 05 42 PM" src="https://user-images.githubusercontent.com/34260251/111667952-97df7300-87eb-11eb-8763-2bc7c36ac108.png">

After script was run –
<img width="839" alt="Screen Shot 2021-03-18 at 1 07 23 PM" src="https://user-images.githubusercontent.com/34260251/111668016-aa59ac80-87eb-11eb-8182-7cc63e68fb1e.png">

SQL Table (after script):
<img width="406" alt="Screen Shot 2021-03-18 at 1 08 15 PM" src="https://user-images.githubusercontent.com/34260251/111668165-dbd27800-87eb-11eb-88ee-c318edf4c3d2.png">

Reply:
good point